tests/system_tests_one_router.py

```diff
@@ -3216,5 +3216,86 @@ def run(self):
                 sleep(0.1)
 
 
+class DataConnectionCountTest(TestCase):
+    """
+    Start the router with different numbers of worker threads and make sure
+    that the automatic setting of the number of data connections works
+    correctly. Also make sure that the user can override that automatic
+    setting by providing a specific integer, including zero.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(DataConnectionCountTest, cls).setUpClass()
+        DataConnectionCountTest.listen_port = cls.tester.get_port()
+        cls.config = Qdrouterd.Config([
+            ('router', {'mode': 'interior', 'id': 'DCC', 'workerThreads': '7'}),
+            ('listener', {'port': DataConnectionCountTest.listen_port, 'role': 'normal'}),
+            ('connector', {'role': 'inter-router', 'host': '127.0.0.1', 'port': DataConnectionCountTest.listen_port, 'saslMechanisms': 'ANONYMOUS', 'idleTimeoutSeconds': '120'}),
+        ])
+
+    def test_60_data_connection_count_absent(self):
+        """
+        If the user omits the dataConnectionCount line from
+        the connector paragraph, it should default to 'auto'.
+        Since this router has worker threads set to '7', we
+        should get (7+1)/2 == 4 data connections.
+        """
+        name = "dcc-test-router"
+        self.router = self.tester.qdrouterd(name, self.config)
+        self.router.wait_ready()
+        msg = "Inter-router data connections calculated at 4"
+        self.router.wait_log_message(msg, timeout=1.0)
```

Comment:
I would remove this timeout=1 parameter. The wait_log_message has its built in timeout and that is a standard timeout we use in all test, so that should suffice.